lib/src/foundation/bloc.dart

```diff
@@ -28,49 +28,104 @@ class Bloc {
     return bloc;
   }
 
+  /// This [StreamController] allows to send events of type [EventType].
+  /// The [DescribedFeatureOverlay]s will be able to handle these events by checking the
+  /// [activeFeatureId] against its own feature id or by considering its current [FeatureOverlayState].
+  final StreamController<EventType> _eventsController =
```

Comment:
While working with this code intensely to get it working in my production app, I realized that we really do not need three seperate streams of events to control the seperate events or actions (open, complete, and dismiss).

@axel-op I am **not** criticizing your decision to go with three individual streams initially because that *was* the way to go and this option of a single stream did not really exist.  
Since `activeFeatureId` is exposed now, it is easily possible to simply send `EventType`s to the `DescribedFeatureOverlay`s, which also has the benefit of not sending unnecessary data.  
The data in an overlay's state combined with the `activeFeatureId` is sufficient to determine if the event should be taken by any particular overlay.  
I know that it is quite invasive to change a lot of code like this, but maybe you can see how I think that this removes a bunch of unnecessary duplication.  
I can now understand why you changed up all the code when introducing the stream implementation initially because ideas like this come up when working with the code this intensely.